gtfs-realtime/proto/gtfs-realtime.proto

```diff
@@ -364,6 +364,24 @@ message TripUpdate {
     extensions 9000 to 9999;
   }
   optional TripProperties trip_properties = 6;
+  
+  // Realtime occupancy immediately after departure for a given stop on a
+  // trip. Occupancies can be supplied for both past and future events.
+  // The producer is allowed, although not required, to drop past events.
+  message StopOccupancy {
+    // The update is linked to a specific stop either through stop_sequence or
+    // stop_id, so one of the fields below must necessarily be set.
+
+    // Must be the same as in stop_times.txt in the corresponding GTFS feed.
+    optional uint32 stop_sequence = 1;
+    // Must be the same as in stops.txt in the corresponding GTFS feed.
+    optional string stop_id = 2;
+
+    // Expected occupancy after departure from the given stop.
+    optional VehiclePosition.OccupancyStatus occupancy_status = 3;
+  }
+
+  optional StopOccupancy stop_occupancy = 7;
```

Comment:
>However then it's necessary to specify time updates in order to specify occupancy updates - this proposal makes times and occupancies independent.

There is precedent for specifying other stop attributes within a `StopTimeUpdate` without specifying predictions. https://github.com/google/transit/pull/219 introduced dynamic stop assignments, which says:

 "To assign a stop without providing any real-time arrival or departure predictions, populate stop_time_properties.assigned_stop_id and set StopTimeUpdate.schedule_relationship = NO_DATA."

We could do something similar here if we want to include the new data within StopTimeUpdate.

I'd be more inclined to bundle the data within StopTimeUpdate and not have a separate array here. Having another array of values that have to be matched to stops and can be keyed on stop_id or stop_sequence seems like it's inviting trouble, as that's already a point of error in feeds. If this stays as a separate array, I'd strongly recommend that it be keyed only on stop_sequence and omit stop_id.

Reply:
Also, this would need to change from `optional` to `repeated` if it stays as a separate array.

Reply:
Thanks. That makes sense to me. I suppose the negatives are having to provide empty `StopTimeUpdate`s in order to provide occupancy predictions (which is probably a niche case), and the name `StopTimeUpdate` possibly implying that it's about time and not occupancy.
They are not show-stoppers to me. Particularly if there is precedent, then a separate repeated looks good. I'll propose some changes after others have had a chance to comment on this and the other points under discussion.